sched/scheduler/task_runner.go

```diff
@@ -29,33 +31,65 @@ func runTaskAndLog(
 	task sched.TaskDefinition,
 	markCompleteOnFailure bool,
 	stat stats.StatsReceiver) error {
-	// Log StartTask Message to SagaLog
-	err := saga.StartTask(taskId, nil)
+
+	// Start Task and then log StartTask message a status payload containing stdout/stderr URIs.
+	// Note: this ordering will result in an orphaned worker if logging fails.
+	//       also, URIs will likely change upon task completion, i.e. from local file to snapshot URI.
+	// TODO: !!this no longer works if we ever start allowing compensating actions on failure!!
```

Comment:
agree I would recommend leave 
err := saga.StartTask(taskId, nil)

make the call to the work
then make the start call to the worker, and then do 
saga.StartTask(taskId, statusAsBytes) 

Also do we want to log the whole ProcessStatus?   Is all the info useful?  We should be careful what we expose in StartTask since clients may take dependencies on this.  I like limiting what we expose to the bare minimum we need